### PR TITLE
Add CONTRIBUTING.md and supporting local setup files

### PR DESCRIPTION
**Pull request by edalex-ian**
We want to make sure it's easy for others to assist contributing the
doco, and we also want to make sure people can preview their work before
they submit it. So I've outlined how that is done in CONTRIBUTING.md and
added the supporting files to be able to run a Jekyll build and site
locally.

**NOTES**

* I'm not 100% sure whether you do need to do the whole CLA thing for the doco repo, but...
* I'm no ruby/gem master, but I what I've provided works for me:tm:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+But after that, as per usual GitHub, please first fork the project and then branch off `master` or a
+stable branch (see below) to use a 'feature' branch.
+
+Please ensure you provide [quality commit messages](https://chris.beams.io/posts/git-commit/),
+and in the PR please detail what testing you have undertaken.
+
+## Setting up a development environment
+
+The site is served up by GitHub pages and is done so via a static [Jekyll](https://jekyllrb.com/)
```
Comment (ChristianMurphy) [inline CONTRIBUTING.md:42]: Using Docker can simplify setup a one liner

```sh
docker run -t --rm -v "$PWD":/usr/src/app -p "4000:4000" starefossen/github-pages
```

https://hub.docker.com/r/starefossen/github-pages

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+## Chatting
+
+If before any of the below you'd like to discuss an issue or some code, then come and chat with
+us on one of the following platforms:
+
+* [Users mailing list](https://groups.google.com/a/apereo.org/forum/#!forum/equella-users) - Most
+  active, best for more usage centric questions
+* [Developer mailing list](https://groups.google.com/a/apereo.org/forum/#!forum/equella-dev) - For
+  discussing work on the code
+* \#Equella channel in [Apereo's slack](https://apereo.slack.com/)
```
Comment (ChristianMurphy) [inline CONTRIBUTING.md]: https://apereo.slack.com/messages/C4KLR0C4U is the direct link to the Equella channel on Slack.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+## Submitting Issues
+
+Issues relating to the documentation are welcome. (For any issues relating to openEQUELLA itself,
+please switch over to its repo.)
+
+When raising issues, please try and be as detailed as possible. Links, line numbers, etc always
```
Comment (ChristianMurphy) [inline CONTRIBUTING.md:24]: It's nice to document this, but most new contributors will ignore this.
Issue templates are an easier way to nudge folks in the right direction.
https://help.github.com/articles/about-issue-and-pull-request-templates
https://help.github.com/articles/creating-issue-templates-for-your-repository

E.G. https://github.com/Jasig/uPortal/tree/master/.github/ISSUE_TEMPLATE

Comment (ChristianMurphy) [reply CONTRIBUTING.md:24]: Something similar to https://github.com/equella/Equella/pull/607 maybe?

File: CONTRIBUTING.md (hunk 1 above)
Comment (edalex-ian) [reply CONTRIBUTING.md:42]: Indeed. But not everyone (especially those doing doco) are going to have (nor want) docker. So I'll leave that for now.

File: CONTRIBUTING.md (hunk 3 above)
Comment (edalex-ian) [reply CONTRIBUTING.md:24]: Being that this is a minor repo (and not even code - so there's not much we can enforce), happy with the soft suggestion. Prefer contribution over process in this one.

File: CONTRIBUTING.md (hunk 2 above)
Comment (edalex-ian) [reply CONTRIBUTING.md]: Nice, I'll go and change that one. The above is just from the Apereo project site.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# Contributing
+
+:+1::tada: First off, thanks for taking the time to contribute! :tada::+1:
+
+We welcome contributions via both the raising of issues and submitting pull requests. But before you
+do, please take a moment to consult the below.
```
Comment (cbeach47) [inline CONTRIBUTING.md]: Consider rewording - "please take a moment to consult the below." > "please take a moment to review the following."

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+## Chatting
+
+If before any of the below you'd like to discuss an issue or some code, then come and chat with
+us on one of the following platforms:
```
Comment (cbeach47) [inline CONTRIBUTING.md]: Confused at the meaning here:  "If before any of the below you'd like to discuss an issue or some code"

File: CONTRIBUTING.md (hunk 3 above)
Comment (cbeach47) [reply CONTRIBUTING.md:24]: I wouldn't say no to a simple issue template PR, but this is sufficient as well.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+Issues relating to the documentation are welcome. (For any issues relating to openEQUELLA itself,
+please switch over to its repo.)
+
+When raising issues, please try and be as detailed as possible. Links, line numbers, etc always
+welcome.
+
+## Contributing Code / Pull Requests
+
+As per standard Apereo projects:  If you wish to contribute to Open EQUELLA itself, you should first
```
Comment (cbeach47) [inline CONTRIBUTING.md]: Open EQUELLA > openEQUELLA

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+subscribe to the Open EQUELLA Google Group as outlined above and discuss your contributions with
```
Comment (cbeach47) [inline CONTRIBUTING.md]: Open EQUELLA > openEQUELLA

File: CONTRIBUTING.md (hunk 1 above)
Comment (cbeach47) [reply CONTRIBUTING.md:42]: Yes, but the same argument could hold for Ruby (possibly to a lesser extent). :)

I would suggest 3 methods for contribution:
small changes - use a text editor and preview in the PR on github
Local preview with Docker - Add Christian's one liner here
Local preview with Ruby - continue with the documentation below

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+the other developers. You will also need to sign a
+[Contributor License Agreement](https://www.apereo.org/node/676).
+
+But after that, as per usual GitHub, please first fork the project and then branch off `master` or a
+stable branch (see below) to use a 'feature' branch.
```
Comment (cbeach47) [inline CONTRIBUTING.md]: Could you reword this? - not clear as to the meaning.  "or a stable branch (see below) to use a 'feature' branch."

**Review by cbeach47 — CHANGES_REQUESTED** [review]
:+1:  Thanks for putting this together @edalex-ian .  Comments inline.

File: CONTRIBUTING.md (hunk 8 above)
Comment (edalex-ian) [reply CONTRIBUTING.md]: Ah yes, that's an issue from bringing across from the equella repo. I'll remove.

File: CONTRIBUTING.md (hunk 1 above)
Comment (edalex-ian) [reply CONTRIBUTING.md:42]: Yeah, definitely to a less extent. Most systems have ruby - well, unless you're on windows.

But sure, I can provide options. I was just trying to keep this quick - as I'm not even entirely sure it'll be read that much. :thinking:

File: CONTRIBUTING.md (hunk 4 above)
Comment (edalex-ian) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 5 above)
Comment (edalex-ian) [reply CONTRIBUTING.md]: Reworded.

File: CONTRIBUTING.md (hunk 6 above)
Comment (edalex-ian) [reply CONTRIBUTING.md]: done - needs doing over in equella/Equella

File: CONTRIBUTING.md (hunk 7 above)
Comment (edalex-ian) [reply CONTRIBUTING.md]: done - needs doing over in equella/Equella

**Comment by edalex-ian** [comment]
BTU @cbeach47

**Comment by cbeach47** [comment]
Sry @edalex-ian , this slipped off my radar.  It's a bit after the fact, but changes look good.  :+1:

**Pull request merged**